### DISCUSSION TITLE
LLM documentation

### DISCUSSION THREAD

**Discussion by anyb1s** [livewire/flux#925] [Feature Requests] [answered]
Can the docs be somehome available to AI editors to use, ie Cursor? Dont think it crawls the pages where requires authentication

**Comment by Hesesses** [comment]
+1

**Comment by danrocha** [comment]
Yes, it would be great to have either the docs as a simple .md or collection of .md files available to download or on a single URL available to scrape.

**Comment by kriiv** [comment]
This would be a massive productivity booster.

**Comment by scarselli** [comment]
+1.  I've even tried scraping the docs and because the code is placed within <code-candy><textarea>, the markdown conversion gets all messed up and I've ultimately been unable to scrape the docs into a markdown file successfully.

This would be a game-changer feature.

**Comment by joshhanley** [comment]
@wishborn sorry I've had to delete your comment. While we appreciate the effort, Flux is a paid product and so redistribution of Flux (and related information) is against the licence agreement, hence why I have deleted your comment. We understand that the limitation of needing to be logged in sucks as LLMs can't fully read the docs. But we also have our reasons why users need to be logged in at this stage.

**Reply by wishborn** [reply]
Everything that was in those docs is publicly available already. All I did was make it easier for the AI to read it.

**Reply by christopherarter** [reply]
> @wishborn sorry I've had to delete your comment. While we appreciate the effort, Flux is a paid product and so redistribution of Flux (and related information) is against the licence agreement, hence why I have deleted your comment. We understand that the limitation of needing to be logged in sucks as LLMs can't fully read the docs. But we also have our reasons why users need to be logged in at this stage.

@joshhanley  I can empathize for sure. Perhaps an LLM friendly text doc behind an auth endpoint for paid users could be a value add and low hanging fruit e.g. `/api/docs/llm?api_token=<xyz>` and easily cacheable so ideally wouldn't incur much resources on Flux side either?

I'd even buy this as a plugin / upgrade to my existing unlimited license.
 
 Just spitballing! 👍

**Reply by joshhanley** [reply]
@wishborn yep, you're more than welcome to do that for your own uses. I'm just asking that you don't share that for the moment. As I said, I get it, I understand. I use LLMs everyday too. But I'm only working off what our policy currently is. We will be reviewing it, but everything takes time.

@christopherarter great idea for sure! Our docs aren't formatted in a way that makes it easy to generate at this stage 😅 the reference docs we've just added are step in that direction though 😁

**Comment by wishborn** [comment]
We are going to see more and more tool and API providers attaching MCP
servers to their docs. This would go a long way towards automated
development.

On Mon, Mar 17, 2025 at 12:05 PM Chris Arter ***@***.***>
wrote:

> @wishborn <https://github.com/wishborn> sorry I've had to delete your
> comment. While we appreciate the effort, Flux is a paid product and so
> redistribution of Flux (and related information) is against the licence
> agreement, hence why I have deleted your comment. We understand that the
> limitation of needing to be logged in sucks as LLMs can't fully read the
> docs. But we also have our reasons why users need to be logged in at this
> stage.
>
> @joshhanley <https://github.com/joshhanley> I can empathize for sure.
> Perhaps an LLM friendly text doc behind an auth endpoint could be a value
> add and low hanging fruit e.g. /api/docs/llm?api_token=<xyz> and easily
> cacheable so ideally wouldn't incur much resources on Flux side either?
>
> I'd even buy this as a plugin / upgrade to my existing unlimited license.
>
> Just spitballing! 👍
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/livewire/flux/discussions/925#discussioncomment-12528182>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADQVBW2C53TYRWNEKKCGVL2U36FJAVCNFSM6AAAAABUETXPNKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTENJSHAYTQMQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Cheers,
-=Wish

**Comment by wishborn** [comment]
yeah no doubt, I wasn't trying to push any particular point. Just mentioned
that as a suggestion on how to serve devs that use this product (and all
products), better. You had mentioned exposing an llm api endpoint, so I
mentioned the MCP server since that is exactly what the purpose of an MCP
server is for, (exposing systems data/knowledge to llms).

P.S. Not assuming you don't know this, I am just being very descriptive for
anyone else that might read this and isn't familiar with the terms.

On Mon, Mar 17, 2025 at 4:02 PM Josh Hanley ***@***.***>
wrote:

> @wishborn <https://github.com/wishborn> yep, you're more than welcome to
> do that for your own uses. I'm just asking that you don't share that for
> the moment. As I said, I get it, I understand. I use LLMs everyday too. But
> I'm only working off what our policy currently is. We will be reviewing it,
> but everything takes time.
>
> @christopherarter <https://github.com/christopherarter> great idea for
> sure! Our docs aren't formatted in a way that makes it easy to generate at
> this stage 😅 the reference docs we've just added are step in that
> direction though 😁
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/livewire/flux/discussions/925#discussioncomment-12530391>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADQVBWSCJFJWMFEHL73ADL2U42AFAVCNFSM6AAAAABUETXPNKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTENJTGAZTSMI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Cheers,
-=Wish

**Comment by scarselli** [comment]
For whatever it may be worth, I'd be willing to throw a few bucks at sponsoring an MCP or just a markdown-formatted version of the docs made for LLM consumption.  We are definitely in different times now with AI code editors, and not having docs easily accessible is a big deal for a lot of us -- as this thread has shown.  So let us know if there is anything we can do to jump start it.

**Comment by kriiv** [comment]
Woohoo! Great news everyone. An update on this front:
"Ungated all code snippets so LLMs can better learn Flux’s API."

**Comment by christopherarter** [comment]
<img width="627" alt="Screenshot 2025-03-27 at 9 05 22 AM" src="https://github.com/user-attachments/assets/1d62f741-4b0e-4e50-8f91-09eafcbd5f6f" />

![28mh3m](https://github.com/user-attachments/assets/f90107bf-d175-4054-b93f-2a33403bc051)

oh bby yes

**Comment by joshhanley** [comment, marked as answer]
Going to close this request as we've now removed all the auth requirements for code snippets.

**Comment by ghbob** [comment]
maybe for future discussions about an MCP server: https://github.com/livewire/flux/discussions/2094

**Reply by wishborn** [reply]
docs for flux are accessible through the laravel boost mcp server so no need for flux to have it's own.

**Reply by ghbob** [reply]
> docs for flux are accessible through the laravel boost mcp server so no need for flux to have it's own.

oh! I didn't know that, thank you!
